### PR TITLE
Add optional capabilities for 4.16 e2e/upgrade

### PR DESCRIPTION
**Pull request by liangxia**
Add two more optional capabilities for 4.16. Also pre-add 4.17.

### CODE REVIEW

File: ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh [hunk 1]
```diff
@@ -149,13 +149,14 @@ function filter_test_by_capability() {
...
     IFS='.' read xversion yversion _ < <(oc version -o yaml | yq '.openshiftVersion')
-    local v411 v412 v413 v414 v415 v416
+    local v411 v412 v413 v414 v415 v416 v417
     v411="baremetal marketplace openshift-samples"
     v412="${v411} Console Insights Storage CSISnapshot"
     v413="${v412} NodeTuning"
     v414="${v413} MachineAPI Build DeploymentConfig ImageRegistry"
     v415="${v414} OperatorLifecycleManager CloudCredential"
-    v416="${v415}"
+    v416="${v415} CloudControllerManager Ingress"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh:158]: As far as I  know, `Ingress` is a bit special, it is always enabled in OCP, while it becomes optional in hypershift cluster.

Comment (liangxia) [reply ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh:158]: This is a list, we use this list to compare with the caps on the cluster to determine which feature to test.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh:158]: nice.

**Pull request merged**